railties/test/application/configuration_test.rb

```diff
@@ -695,12 +695,14 @@ def assert_utf8
       assert_equal Pathname.new(app_path).join("somewhere"), Rails.public_path
     end
 
-    test "In production mode, config.public_file_server.enabled is on by default" do
+    test "In all modes, config.public_file_server.enabled is on by default" do
       restore_default_config
 
-      with_rails_env "production" do
-        app "production"
-        assert app.config.public_file_server.enabled
+      %w[development production test].each do |environment|
+        with_rails_env environment do
+          app environment
+          assert app.config.public_file_server.enabled
+        end
```

Comment:
I'm not sure how well this loop works with respect to test setup and teardown behavior.  If we want to test the other environments, I think we would need separate tests.

Reply:
I see, I split the test loop into 3 tests